hera_cal/tempcal.py

```diff
@@ -0,0 +1,203 @@
+            flags : DataContainer
+                Holds data flags of shape (Ntimes, Nflags)
+            smooth_frate : float
+                Fringe-Rate cutoff for smoothing in mHz
+            nl : float
+                Noise level in GP interpolation.
+            Nmirror : int
+                Number of time bins to mirror about edges before interpolation
+            keys : list
+                List of ant-pair-pol tuples to operate on
+            edgeflag : int or len-2 tuple
+                Number of channels to flag on low and high end of frequency band.
+                Low and high can be specified independently via a len-2 tuple.
+            verbose : bool
+                If True, report feedback to stdout.
+        """
+        # get flags
+        if flags is None:
+            flags = DataContainer(dict([(k, np.zeros_like(data[k], np.bool)) for k in data]))
+
+        # edgeflag
+        if edgeflag is not None:
+            flags = copy.deepcopy(flags)
+            if isinstance(edgeflag, (int, np.int, float, np.float)):
+                edgeflag = (edgeflag, edgeflag)
+            assert len(edgeflag) == 2
+            for k in flags:
+                if edgeflag[0] > 0:
+                    flags[k][:, :edgeflag[0]] = True
+                if edgeflag[1] > 0:
+                    flags[k][:, -edgeflag[1]:] = True
+
+        # get length scale in seconds
+        length_scale = 1.0 / (smooth_frate * 1e-3) / (24.0 * 3600.0)
+
+        # get keys
+        if keys is None:
+            keys = list(data.keys())
+
+        # smooth data
+        if not hasattr(self, 'smooth'):
+            self.smooth = DataContainer({})
+        if not hasattr(self, 'ratio'):
+            self.ratio = DataContainer({})
+        for k in keys:
+            # smooth data
+            utils.echo("working on {}".format(k), verbose=verbose)
+            gkey = utils.split_bl(k)[0]
+            data_shape = data[k].shape 
+            self.smooth[k] = utils.gp_interp1d(times, data[k], flags=flags[k], length_scale=length_scale, nl=nl,
+                                               Nmirror=Nmirror)
+            self.gflags[gkey] = np.isclose(self.smooth[k], 0.0)
+
+            # take ratio and compute gain term
+            self.ratio[k] = np.true_divide(self.data[k], self.smooth[k], where=~self.gflags[gkey])
+            self.gains[gkey] = np.sqrt(self.ratio[k])
+
+            # only allow frequency-averaged gains for now via nanmedian
+            # use median over mean to help w/ unflagged RFI
+            g = self.gains[gkey].copy()
+            g[self.gflags[gkey]] = np.nan
+            self.gains[gkey] = np.nanmedian(g, axis=1, keepdims=True)
+            self.gains[gkey][np.isnan(self.gains[gkey])] = 1.0  # catch for fully flagged integrations
+            self.gains[gkey] = np.repeat(self.gains[gkey], data_shape[1], axis=1)
+            self.gflags[gkey] = np.repeat(np.min(self.gflags[gkey], axis=1, keepdims=True), data_shape[1], axis=1)
```

Comment:
I'd use np.all() rather than np.min(). It's more much readable since it's clear that you're working with bools.